### DISCUSSION TITLE
Xrdp Slow ?

### DISCUSSION THREAD

**Discussion by gamersalpha** [neutrinolabs/xrdp#2136] [Q&A]
Hello, 

well i m ussing XRDP on ubuntu desktop

but wheni run it the window are slow, how can i get a better flow please ?

look when i move the windows, is too slow
![mstsc_lOjZiODQhA](https://user-images.githubusercontent.com/32466950/151656201-9909b0c6-94cc-46d6-97fb-a921ccb4fc7e.gif)

how to optimise the XRDP please ?

and i still not got sound !

**Comment by matt335672** [comment]
The best you can do with recent versions is to enable the RFX codec, if you're using a suitable network.

To do this, make sure 'LAN connection' is selected on your RDP client. Here's an example for mstsc.exe:-
![image](https://user-images.githubusercontent.com/30179339/151766724-52ca8ea9-440b-4b88-bff4-6cca7c45bbe1.png)

To get sound you'll need to compile the [pulseaudio module](https://github.com/neutrinolabs/pulseaudio-module-xrdp) yourself. This is now a little easier on recent Ubuntu/Debian versions.

**Reply by nahirrebeg** [reply]
this is a magic bullet - thank you so much @matt335672

**Reply by ming-hao-xu** [reply]
How to do this on macOS?

**Reply by ericemmrich** [reply]
Is there any way to enforce the RFX codec in `xrdp`?

**Reply by brandonscloud** [reply]
@matt335672 Thanks for this. My Kali VM was really lagging on input over RDP, even though I have a really high wired and wireless network. This did the trick.

**Reply by Kreijstal** [reply]
Why does windows on windows is fast without needing this? does it not use the RFX codec? Does it automatically detect it?

**Comment by peddanet** [comment]
Great Tip! Now the sound is quite ok! And the speed is ok, most of the time. Now my only problem is that gnome keyring has to be unlocked the first time  I logged into a new session, also if I put in a USB stick (though I thought I have adapted the mighty polkit rules. But anyway most of my issues are gone away. I am asking myself why this is such a hobbyistic way....

**Reply by akarl10** [reply]
for gnome keyring: you might try to change your /etc/pam.d/xrdp-sesman to something like
```
#%PAM-1.0
auth required pam_env.so readenv=1
auth required pam_env.so readenv=1 envfile=/etc/default/locale
@include common-auth
-auth   optional        pam_gnome_keyring.so
-auth   optional        pam_kwallet5.so
@include common-account
@include common-session
-session optional       pam_gnome_keyring.so auto_start
-session optional       pam_kwallet5.so auto_start
@include common-password
```

if you don't have kwallet, just ignore those lines.
just add the lines about in that spot in your existing file (with the - in front)

**Comment by mega-optimus** [comment]
I remote from Windows to Linux within LAN environment, where Linux box is connected via a wireless adapter. 
I experience slowliness with latest xrdp as well. But eventually I figured out the cause is elsewhere. 
After I turned off the adapter power-saving in /etc/NetworkManager/conf.d/default-wifi-powersave-on.conf, slowliness and random disconnect goes away ....
